include/linuxdeploy/core/appdir.h

```diff
@@ -26,7 +26,7 @@ namespace linuxdeploy {
                     // default constructor
                     // construct AppDir from given path
                     // the directory will be created if it doesn't exist
-                    explicit AppDir(const boost::filesystem::path& path);
+                    explicit AppDir(const boost::filesystem::path& path, const std::vector<std::string> &excludeLibraryPatterns);
```

Comment:
Please use a setter for those patterns. Passing empty lists everywhere makes no sense, really.